ocaml/typing/env.ml

```diff
@@ -2241,6 +2241,8 @@ let add_value_lazy ?check ?shape ?(mode=Mode.Value.allow_right Mode.Value.legacy
   store_value ?check mode id addr desc shape env
 
 let add_type ~check ?shape id info env =
+  (* CR layouts: there should be a safety check for extension universe when the type's
```

Comment:
Why? I think doing this check here is awkward. I'd rather do in jkind when we interpret the user-written syntax. I guess I'm depending on the fact that mode-crossing can't suddenly arise without such an annotation.

Reply:
Yes I think you're right that this CR is wrong. I can't even remember why I wrote it in the first place.

Reply:
And this check is already done when interpreting user syntax.